src/snowflake/snowpark/session.py

```diff
@@ -330,6 +337,12 @@ def sql_simplifier_enabled(self, value: bool) -> None:
         self._conn._telemetry_client.send_sql_simplifier_telemetry(
             self._session_id, value
         )
+        try:
+            self._conn.run_query(
+                f"alter session set {_PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING} = {value}"
```

Comment:
Just a quick note that this won't work inside owner's right stored proc. Should be okay since this is just a setter and not on the critical path, and for stored proc they could `alter session` before calling the stored proc.

Reply:
this will also not work on our merge gate (in production), because the role doesn't have the privilege to alter the parameter, so does for customers. So I guess here just changing the local attribute is ok.